### PR TITLE
Plane: quadplane and tiltrotor improvements

### PR DESCRIPTION
**Pull request by tridge**
This is a set of incremental improvements based on log analysis of real aircraft.
It improves logging so we can see the quadplane and fixed wing PIDs separately.
It makes the choice of initial integrator for the Z controller smarter
It improves the logic for tiltrotor compensation to be asymmetric, based on target mode

### CODE REVIEW

File: ArduPlane/tiltrotor.cpp [hunk 1]
```diff
@@ -198,14 +198,23 @@ void QuadPlane::tiltrotor_update(void)
 
   By applying _tilt_equal_thrust the tilted motors effectively become
   a single pitch control motor.
+
+  Note that we use a different strategy for when we are transitioning
+  to VTOL as compared to from VTOL flight. The reason for that is we
```
Comment (OXINARF) [inline ArduPlane/tiltrotor.cpp]: Nitpicking: the sentence might look better as _Note that we use a different strategy for when we are transitioning **into** VTOL as compared to from VTOL flight_. At least it was confusing to me reading the way it is now.

**Comment by tridge** [comment]
thanks for the review @OXINARF! pushed with your suggested wording change

**Comment by magicrub** [comment]
We really really really need to sit down one day and document the logging structure. All the platforms look so different. I wish we could self describe this data like the params so the tools and wiki are always up to date.

**Comment by peterbarker** [comment]
On Tue, 14 Mar 2017, Tom Pittenger wrote:

> We really really really need to sit down one day and document the logging
> structure. All the platforms look so different. I wish we could self
> describe this data like the params so the tools and wiki are always up to
> date.

They're not set in stone, and they'll drift.

This was an attempt to formalise at least the units part of things:
https://github.com/ArduPilot/ardupilot/compare/master...peterbarker:units?expand=1

... but even when I was trying to maintain that it was a PITA.

**Comment by magicrub** [comment]
I understand, that's why a self updating method is what we should aim for (like params).

**Pull request closed without merging**